domain/model/watcher_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package model_test
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/changestream"
+	"github.com/juju/juju/core/database"
+	"github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/watcher/watchertest"
+	"github.com/juju/juju/domain"
+	"github.com/juju/juju/domain/model/service"
+	"github.com/juju/juju/domain/model/state"
+	modeltesting "github.com/juju/juju/domain/model/state/testing"
+	changestreamtesting "github.com/juju/juju/internal/changestream/testing"
+	loggertesting "github.com/juju/juju/internal/logger/testing"
+)
+
+type watcherSuite struct {
+	changestreamtesting.ControllerSuite
+}
+
+var _ = gc.Suite(&watcherSuite{})
+
+func Test(t *testing.T) {
+	gc.TestingT(t)
+}
+
+type dbModel struct {
+	UUID        string `db:"uuid"`
+	Activated   bool   `db:"activated"`
+	ModelTypeID int    `db:"model_type_id"`
+
+	Name      string `db:"name"`
+	CloudUUID string `db:"cloud_uuid"`
+	LifeID    int    `db:"life_id"`
+	OwnerUUID string `db:"owner_uuid"`
+}
+
+func (s *watcherSuite) TestWatchControllerDBModels(c *gc.C) {
+	logger := loggertesting.WrapCheckLog(c)
+	watchableDBFactory := changestream.NewWatchableDBFactoryForNamespace(s.GetWatchableDB, "model")
+	watcherFactory := domain.NewWatcherFactory(watchableDBFactory, logger)
+	st := state.NewState(func() (database.TxnRunner, error) { return watchableDBFactory() })
+	modelSvc := service.NewWatchableService(st, nil, loggertesting.WrapCheckLog(c), watcherFactory)
+
+	// Create a controller service watcher.
+	watcher, err := modelSvc.WatchActivatedModels(context.Background())
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(watcher, gc.NotNil)
```

Comment:
This isn't needed

Reply:
Hi Tom, pmed you about this, not sure which line(s) you mean here but currently removed "c.Assert(watcher, gc.NotNil)"